client/solve.go

```diff
@@ -158,8 +165,27 @@ func (c *Client) solve(ctx context.Context, def *llb.Definition, runGateway runG
 			}
 		}
 
-		if len(cacheOpt.contentStores) > 0 {
-			s.Allow(sessioncontent.NewAttachable(cacheOpt.contentStores))
+		// this is a new map that contains both cacheOpt stores and OCILayout stores
+		contentStores := map[string]content.Store{}
```

Comment:
initialize with the length that is the sum of two previous maps.

Reply:
Sure, why not?